browser_context.go

```diff
@@ -210,21 +214,57 @@ func (b *browserContextImpl) ExposeFunction(name string, binding ExposedFunction
 }
 
 func (b *browserContextImpl) Route(url interface{}, handler routeHandler, times ...int) error {
-	b.routes = append(b.routes, newRouteHandlerEntry(newURLMatcher(url, b.options.BaseURL), handler, times...))
+	b.Lock()
+	defer b.Unlock()
+	b.routes = slices.Insert(b.routes, 0, newRouteHandlerEntry(newURLMatcher(url, b.options.BaseURL), handler, times...))
 	return b.updateInterceptionPatterns()
 }
 
 func (b *browserContextImpl) Unroute(url interface{}, handlers ...routeHandler) error {
+	removed, remaining, err := unroute(b.routes, url, handlers...)
+	if err != nil {
+		return err
+	}
+	return b.unrouteInternal(removed, remaining, UnrouteBehaviorDefault)
+}
+
+func (b *browserContextImpl) unrouteInternal(removed []*routeHandlerEntry, remaining []*routeHandlerEntry, behavior *UnrouteBehavior) error {
 	b.Lock()
 	defer b.Unlock()
-
-	routes, err := unroute(b.routes, url, handlers...)
+	b.routes = remaining
+	err := b.updateInterceptionPatterns()
 	if err != nil {
```

Comment:
```suggestion
	if err := b.updateInterceptionPatterns(); err != nil {
```

Reply:
Hi @mxschmitt , I'm considering forcing the use of `gofumpt` to format the code, which would avoid these. What do you think?

Reply:
Feel free to add!